projects/organization-management/src/app/services/users/users.service.ts

```diff
@@ -17,4 +20,64 @@ export class UsersService {
       map(users => users.map(UserMapper.fromData))
     );
   }
+
+  deleteUser(customerItemUserKey: string) {
+    if (!customerItemUserKey) {
+      return throwError('deleteUser() called without customerItemUserKey');
+    }
+
+    return this.apiService.delete(`customers/-/users/${customerItemUserKey}`);
+  }
+
+  addUser(body: CustomerUserType): Observable<User> {
+    if (!body || !body.customer || !body.user) {
+      return throwError('addUser() called without required body data');
+    }
+
+    if (!body.customer.type) {
+      return throwError('addUser() called without required customer type (PrivateCustomer/SMBCustomer)');
+    }
+
+    return this.apiService
+      .post<User>(`customers/${body.customer.customerNo}/users`, {
+        type: 'SMBCustomerUserCollection',
+        name: 'Users',
+        elements: [
+          {
+            ...body.customer,
+            ...body.user,
+            preferredInvoiceToAddress: { urn: body.user.preferredInvoiceToAddressUrn },
+            preferredShipToAddress: { urn: body.user.preferredShipToAddressUrn },
+            preferredPaymentInstrument: { id: body.user.preferredPaymentInstrumentId },
+            preferredInvoiceToAddressUrn: undefined,
+            preferredShipToAddressUrn: undefined,
+            preferredPaymentInstrumentId: undefined,
+          },
+        ],
+      })
+      .pipe(map(UserMapper.fromData));
+  }
+
+  updateUser(customerItemUserKey: string, body: CustomerUserType): Observable<User> {
+    if (!body || !body.customer || !body.user) {
+      return throwError('updateUser() called without required body data');
+    }
+
+    if (!body.customer.type) {
+      return throwError('updateUser() called without required customer type (PrivateCustomer/SMBCustomer)');
+    }
+
+    return this.apiService
+      .put<User>(`customers/${body.customer.customerNo}/users/${customerItemUserKey}`, {
+        ...body.customer,
+        ...body.user,
+        preferredInvoiceToAddress: { urn: body.user.preferredInvoiceToAddressUrn },
+        preferredShipToAddress: { urn: body.user.preferredShipToAddressUrn },
+        preferredPaymentInstrument: { id: body.user.preferredPaymentInstrumentId },
+        preferredInvoiceToAddressUrn: undefined,
+        preferredShipToAddressUrn: undefined,
+        preferredPaymentInstrumentId: undefined,
+      })
+      .pipe(map(UserMapper.fromData));
```

Comment:
resolveLink is missing

Reply:
is this valid for the update user? create user returns a different object in comparison to update user. lets discuss it monday

Reply:
you are right, create user (post request) returns a link, whereas update user (put request) returns the user, really strange, so the service has to handle these differences